packages/@wdio_electron-utils/src/index.ts

```diff
@@ -296,13 +296,85 @@ export async function getAppBuildInfo(pkg: NormalizedReadResult): Promise<AppBui
   throw new Error(BUILD_TOOL_DETECTION_ERROR);
 }
 
-export function getElectronVersion(pkg: NormalizedReadResult) {
+type PnpmWorkspace = { catalog?: { [key: string]: string }; catalogs?: { [key: string]: string } };
+
+export async function findPnpmCatalogVersions(
+  pkgElectronVersion?: string,
+  pkgElectronNightlyVersion?: string,
+  projectDir?: string,
+) {
+  if (!projectDir) {
+    return undefined;
+  }
+
+  // Determine catalog names
+  const electronCatalogName = pkgElectronVersion?.split('catalog:')[1]?.trim();
+  const electronNightlyCatalogName = pkgElectronNightlyVersion?.split('catalog:')[1]?.trim();
+
+  try {
+    // Traverse up the directory tree to find pnpm-workspace.yaml
+    let currentDir = projectDir;
+    let workspaceYamlPath;
+    let yamlContent;
+
+    while (currentDir !== path.parse(currentDir).root) {
+      workspaceYamlPath = path.join(currentDir, 'pnpm-workspace.yaml');
+      try {
+        yamlContent = await fs.readFile(workspaceYamlPath, 'utf8');
+        log.debug(`Found pnpm-workspace.yaml at ${workspaceYamlPath}`);
+        break;
+      } catch (_e) {
+        // Move up one directory
+        currentDir = path.dirname(currentDir);
+      }
+    }
+
+    if (!yamlContent) {
+      return undefined;
+    }
+
+    const pnpmWorkspace = (await import('yaml')).parse(yamlContent) as PnpmWorkspace;
+
+    // Check for electron with named catalog
+    if (electronCatalogName && pnpmWorkspace.catalogs?.[electronCatalogName]) {
+      return pnpmWorkspace.catalogs[electronCatalogName];
```

Comment:
Considering into the [docs](https://pnpm.io/en/catalogs), is this correct electron version at the default catalog?
```suggestion
      return pnpmWorkspace.catalogs[electronCatalogName].electron;
```